packages/react-components/react-aria/src/activedescendant/useOptionWalker.ts

```diff
@@ -0,0 +1,90 @@
+import * as React from 'react';
+import { useFluent_unstable as useFluent } from '@fluentui/react-shared-contexts';
+import { isHTMLElement, useIsomorphicLayoutEffect } from '@fluentui/react-utilities';
+
+interface UseOptionWalkerOptions {
+  matchOption: (el: HTMLElement) => boolean;
+}
+
+export function useOptionWalker<TListboxElement extends HTMLElement>(options: UseOptionWalkerOptions) {
+  const { matchOption } = options;
+  const { targetDocument } = useFluent();
+  const treeWalkerRef = React.useRef<TreeWalker | null>(null);
+  const listboxRef = React.useRef<TListboxElement | null>(null);
+
+  const optionFilter = React.useCallback(
+    (node: Node) => {
+      if (!isHTMLElement(node)) {
+        return NodeFilter.FILTER_SKIP;
+      }
+
+      return matchOption(node) ? NodeFilter.FILTER_ACCEPT : NodeFilter.FILTER_SKIP;
```

Comment:
```suggestion
      if (isHTMLElement(node) && matchOption(node)) {
        return NodeFilter.FILTER_ACCEPT;
      }

      return NodeFilter.FILTER_SKIP;
```

nit: IMO it's a bit clearer